packages/core/src/connectivity/scp-cf/destination/destination-accessor-authentication-type.spec.ts

```diff
@@ -181,6 +183,68 @@ describe('authentication types', () => {
     });
   });
 
+  describe('authentication type OAuth2ClientCredentials', () => {
+    it('returns a destination with authTokens if its authenticationType is OAuth2ClientCredentials, subscriber tenant', async () => {
+      mockServiceBindings();
+      mockVerifyJwt();
+      mockServiceToken();
+      mockUserApprovedServiceToken();
+
+      const httpMocks = [
+        mockInstanceDestinationsCall(nock, [], 200, subscriberServiceToken),
+        mockSubaccountDestinationsCall(
+          nock,
+          oauthClientCredentialsMultipleResponse,
+          200,
+          subscriberServiceToken
+        ),
+        mockSingleDestinationCall(
+          nock,
+          oauthClientCredentialsSingleResponse,
+          200,
+          destinationName,
+          wrapJwtInHeader(subscriberServiceToken).headers
+        )
+      ];
+
+      const expected = parseDestination(oauthClientCredentialsSingleResponse);
+      const actual = await getDestination(destinationName, {
+        userJwt: subscriberUserJwt
+      });
+      expect(actual).toMatchObject(expected);
+      httpMocks.forEach(mock => expect(mock.isDone()).toBe(true));
+    });
+  });
+
+  it('returns a destination with authTokens if its authenticationType is OAuth2ClientCredentials, provider tenant', async () => {
+    mockServiceBindings();
+    mockVerifyJwt();
+    mockServiceToken();
+    mockUserApprovedServiceToken();
+
+    const httpMocks = [
+      mockInstanceDestinationsCall(nock, [], 200, providerServiceToken),
+      mockSubaccountDestinationsCall(
+        nock,
+        oauthClientCredentialsMultipleResponse,
+        200,
+        providerServiceToken
+      ),
+      mockSingleDestinationCall(
```

Comment:
Please correct me, I saw the `xxxSingleResonse` and `xxxMultipleResonse` are mocked together in multiple places.
I guess this is because we as sdk are doing optimisation and fetch all the potential destinations from a provider/subscriber token and cache them.
If I'm an application developer, I can avoid the additional call and just ge the single response, right?

Reply:
No you can't. I have already created a refactoring ticket for 2.0 to get rid of this premature optimization. Since for all OAuth flows you need to call the find destination by name anyhow the gain of fetching all destinations first is not big and it is making the code unnecessary complicated.